apps/designer/app/designer/features/style-panel/style-source/style-source-input.tsx

```diff
@@ -259,13 +306,50 @@ export const StyleSourceInput = <Item extends IntermediateItem>(
           <ComboboxListbox {...getMenuProps()}>
             {isOpen &&
               items.map((item, index) => {
+                if (item.id === newItemId) {
+                  hasNewTokenItem = true;
```

Comment:
a bit dirty setting the flag in the same loop, logic probably would be cleaner if you separate nicely things you need to do for new token item

Reply:
To keep downshift working it should be single list

Reply:
sure, I meant it more from code organization perspective, in the end still a single list, idk, side effects in map isn't very nice

Reply:
The point is this list comes from combobox. The item is added by the logic invoked internally. Splitting would require 2 filters additionally to map and adjusting index. I don't see it make code simpler.